tfsdk/schema.go

```diff
@@ -75,40 +111,32 @@ func (s Schema) AttributeTypeAtPath(path *tftypes.AttributePath) (attr.Type, err
 		return nil, fmt.Errorf("%v still remains in the path: %w", remaining, err)
 	}
 
-	typ, ok := rawType.(attr.Type)
-	if ok {
+	switch typ := rawType.(type) {
+	case attr.Type:
 		return typ, nil
-	}
-
-	if n, ok := rawType.(nestedAttributes); ok {
-		return n.AttributeType(), nil
-	}
-
-	if s, ok := rawType.(Schema); ok {
-		return s.AttributeType(), nil
-	}
-
-	a, ok := rawType.(Attribute)
-	if !ok {
+	case nestedAttributes:
+		return typ.AttributeType(), nil
+	case nestedBlock:
+		return typ.Block.attributeType(), nil
+	case Attribute:
+		return typ.attributeType(), nil
+	case Block:
+		return typ.attributeType(), nil
+	case Schema:
+		return typ.AttributeType(), nil
+	default:
```

Comment:
This whole block makes me think we're missing a chance here for an interface. I don't think that refactoring needs to happen as part of this PR. Just calling it out as something to think on, and at most maybe add a TODO about.

Reply:
Created https://github.com/hashicorp/terraform-plugin-framework/issues/223 👍

Reply:
Thanks!